src/core/test-unit/Reflection/MemberInfoGetTypeOfValueExtensionsTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using RapidCore.Reflection;
+using Xunit;
+
+namespace RapidCore.UnitTests.Reflection
+{
+    public class MemberInfoGetTypeOfValueExtensionsTests
+    {
+        private int MyField = 666;
+        private string MyProp => "Sup Lucifer";
+
+        public MemberInfoGetTypeOfValueExtensionsTests()
+        {
+            // just here for testing purposes
+        }
+        
+        [Fact]
+        public void GetTypeOfValue_onFieldInfo_works()
+        {
+            var memberInfo = GetField("MyField");
+            
+            Assert.Equal(typeof(int), memberInfo.GetTypeOfValue());
```

Comment:
Just for the fun of it, it might be good with a test of nullable value types, like `int?` and so on. It probably works fine, but it doesen't hurt to test it.

Reply:
I did test some of these things with nullables and they actually behave just like regular types, but I will add tests.

Reply:
They do actually come out as the nullable type - I have previously experienced them throwing the "nullable" information away.

Anywhoo.. we now have tests :)